crates/taralli-server/src/postgres.rs

```
@@ -0,0 +1,181 @@
+use crate::error::{Result, ServerError};
+use alloy::primitives::B256;
+use chrono::{DateTime, Utc};
+use deadpool_postgres::{Manager, Pool};
+use serde::Serialize;
+use taralli_primitives::{
+    intents::ComputeOffer,
+    systems::{ProvingSystem, ProvingSystemId},
+    utils::compute_offer_id,
+};
+use tokio_postgres::{Config, NoTls, Row, Transaction};
+
+pub const INSERT_OFFER: &str = "
+    INSERT INTO offers (offer_id, proving_system_id, proving_system, proof_offer, signature, expiration_timestamp, created_at, expired_at)
+    VALUES ($1, $2, $3, $4, $5, $6, $7, $8)
+    RETURNING offer_id, proving_system_id, proving_system, proof_offer, signature, expiration_timestamp, created_at, expired_at;
+";
+
+pub const UPDATE_EXPIRED_OFFERS: &str = "
+    UPDATE offers
+    SET = expired_at = NOW()
+    WHERE offers.expiration_timestamp < NOW()
+    AND expired_at IS NULL
+    RETURNING offer_id, proving_system_id, proving_system, proof_offer, signature, expiration_timestamp, created_at, expired_at;
+";
+
+pub const GET_OFFER_BY_ID: &str = "
+    SELECT * FROM offers
```

rule of thumb is to just not use `*` in case any breaking change comes through, it is better to fail explicitly